pkg/scanners/scannerv4/scannerv4.go

```diff
@@ -202,3 +208,92 @@ func (s *scannerv4) GetVulnerabilities(image *storage.Image, components *types.S
 
 	return imageScan(image.GetMetadata(), vr), nil
 }
+
+func (s *scannerv4) GetNodeVulnerabilityReport(node *storage.Node, indexReport *v4.IndexReport) (*v4.VulnerabilityReport, error) {
+	nodedigest, err := name.NewDigest("registry/repository@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
+	if err != nil {
+		log.Errorf("Failed to parse digest from node %q: %v", node.GetName(), err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), scanTimeout)
+	defer cancel()
+
+	vr, err := s.scannerClient.GetVulnerabilities(ctx, nodedigest, indexReport.GetContents())
+	if err != nil {
+		log.Errorf("Failed to create vulnerability report: %v", err)
+		return nil, errors.Wrap(err, "Failed to create vulnerability report")
```

Comment:
I saw this pattern in few places - we log the error, wrap it and return it. Somewhere up the stack, the error will be printed (including all the wrappings), so maybe the additional logging is not necessary? 
If the log produces some more information, the it could stay, but in this example it prints the same thing as passed to the Wrap function.

Reply:
unresolved